src/crimson/os/seastore/journal.cc

```diff
@@ -147,6 +147,10 @@ ceph::bufferlist Journal::encode_record(
     sizeof(checksum_t),
     reinterpret_cast<const char *>(&metadata_crc_le));
 
+  // writev requires each buffer to be aligned to the disks' block
+  // size, we need to rebuild here
+  bl.rebuild_aligned(block_size);
```

Comment:
Is there a way to build `bl` in an aligned way, so we don't need to `rebuild_aligned()`?

Reply:
Seems that list::reserve() can help, will try

Reply:
It seems that `list::reserve()` is not effective in our case, in which we have to encode other bufferlists in the current bl.

Reply:
It looks to me that the current bufferlist implementation doesn't support what we are proposing